orderBook.js

```diff
@@ -0,0 +1,43 @@
+const reconcileOrder = (existingBook, incomingOrder) =>
+  existingBook.length
+    ? checkAllTypesMatch(existingBook, incomingOrder)
+    : addToBook(existingBook, incomingOrder)
+
+const checkAllTypesMatch = (existing, incoming) =>
+  existing.every((obj) => obj.type === incoming.type)
+    ? addToBook(existing, incoming)
+    : checkPricesMatch(existing, incoming)
+
+const checkPricesMatch = (existing, incoming) => {
```

Comment:
This function is responsible for checking and then also for flow of control after that check just like the others. You're also looping inside a loop since this function has a for loop and is called from within an every.